apache/superset
# perf(dashboard): Improve perf of highlighting charts in scope of active filter

### SUMMARY
Due to passing as a prop `nativeFilters` object (from redux), which changes on each focus/unfocus of native filters, from `DashboardComponent`, all dashboard components (tabs, rows, chart holders, dividers...) were being rerendered on each native filter focus. This PR optimizes that behaviour by passing `nativeFilters` only to `Tabs` component.
Also, I optimized rendering `FiltersBadge` component - due to passing complex objects as props, expensive calculations were being re-run multiple times.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS
0. Enable `DASHBOARD_NATIVE_FILTERS` feature flag
1. Open a dashboard and add some native filters
2. Focus on a native filter, verify that charts and tabs highlighting works as before (but it should be a bit faster on big dashboards, as only `FilterFocusHighlight` component gets rerendered).

### ADDITIONAL INFORMATION



CC: @graceguo-supercat @villebro @junlincc
